javatests/arcs/core/analysis/testdata/ok_field_list_direct.arcs

```diff
@@ -0,0 +1,27 @@
+// #Ingress: P1
+// #OK
+schema FooList
```

Comment:
Can we have a test that verifies that reading a safe subset is ok?

E.g.

This is a PASS:
```
particle P1
  foo: writes FooList {
    elements: List<inline Foo {
      ok: Text,
      notOk: Number,
      mixed: inline Baz {
        ok: Text,
        notOk: Number,
      }
    }> 
  }
  claim foo.elements.ok is trusted
  claim foo.elements.mixed.ok is trusted

particle P2
  foo: reads FooList {
    elements: List<inline Foo {
      ok: Text,
      mixed: inline Baz {ok: Text}
    }>
  }
  check foo is trusted

recipe R
  P1
    foo: writes h
  P2
    foo: reads h
```

This is a FAIL:
```
particle P1
  foo: writes FooList {
    elements: List<inline Foo {
      ok: Text,
      notOk: Number,
      mixed: inline Baz {
        ok: Text,
        notOk: Number,
      }
    }> 
  }
  claim foo.elements.ok is trusted
  claim foo.elements.notOk is untrusted
  claim foo.elements.mixed.ok is trusted
  claim foo.elements.mixed.notOk is untrusted

particle P2
  foo: reads FooList {
    elements: List<inline Foo {
      ok: Text,
      mixed: inline Baz {
        notOk: Number,
      }
    }>
  }
  check foo is trusted

recipe R
  P1
    foo: writes h
  P2
    foo: reads h
```

Reply:
yes. added these as additional tests.